### ISSUE TITLE
Run `podman machine init` as root, and /etc/containers/containers.conf is created with permissions of 600.

### ISSUE THREAD

**Issue by tnk4on** [containers/podman#10732]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

First, when I run `podman machine init` as root, `/etc/containers/containers.conf` is created with permission of 600.
Second, when I run any `podman` command as non-root, I get a permission denied error.

**Steps to reproduce the issue:**

1. Run `podman machine init` as root.

```
# podman machine init
Extracting compressed file
# ls -l /etc/containers/containers.conf
-rw-------. 1 root root 602  6月 20 03:25 /etc/containers/containers.conf
```

2. Run any `podman` command as non-root, I get a permission denied error.

```
$ podman version
ERRO[0000] reading system config "/etc/containers/containers.conf": decode configuration /etc/containers/containers.conf: open /etc/containers/containers.conf: permission denied
```

**Describe the results you received:**

```
$ podman version
ERRO[0000] reading system config "/etc/containers/containers.conf": decode configuration /etc/containers/containers.conf: open /etc/containers/containers.conf: permission denied
```

**Describe the results you expected:**

`/etc/containers/containers.conf` will be created with the correct permissions. Any `podman` command can be executed by a non-root user.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      3.2.1
API Version:  3.2.1
Go Version:   go1.16.3
Built:        Tue Jun 15 04:12:29 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.21.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.29-1.fc34.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: '
  cpus: 2
  distribution:
    distribution: fedora
    version: "34"
  eventLogger: journald
  hostname: fedora34-server
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.12.8-300.fc34.x86_64
  linkmode: dynamic
  memFree: 1451524096
  memTotal: 4114993152
  ociRuntime:
    name: crun
    package: crun-0.20.1-1.fc34.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.20.1
      commit: 0d42f1109fd73548f44b01b3e84d04a279e99d2e
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 3727159296
  swapTotal: 4114608128
  uptime: 213h 59m 22.27s (Approximately 8.88 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 3
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.1
  Built: 1623697949
  BuiltTime: Tue Jun 15 04:12:29 2021
  GitCommit: ""
  GoVersion: go1.16.3
  OsArch: linux/amd64
  Version: 3.2.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.2.1-1.fc34.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

- Fedora release 34 (Thirty Four)
- VM on VMware ESXi

**Comment by vrothberg** [comment]
Thanks for opening the issue, @tnk4on!

@ashley-cui, can you have a look?  Why is Podman creating containers.conf?

**Comment by ashley-cui** [comment]
Huh interesting bug. I'm able to reproduce, will dig into it now!

**Comment by rhatdan** [comment]
We should not create that file unless content is being added, it and should be 644, if created.

**Comment by vrothberg** [comment]
This has been fixed in master and is backported via https://github.com/containers/podman/pull/10763 to the soon to be released v3.2.2.

**Comment by tnk4on** [comment]
I confirmed the fix in v3.2.2. Thank you!
